community/detectors/laravel_cve_2021_3129/src/test/java/com/google/tsunami/plugins/detectors/rce/cve20213129/Cve20213129VulnDetectorTest.java

```diff
@@ -0,0 +1,172 @@
+                    Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                    Vulnerability.newBuilder()
+                        .setMainId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("TSUNAMI_COMMUNITY")
+                                .setValue("CVE_2021_3129"))
+                        .setSeverity(Severity.CRITICAL)
+                        .setTitle("CVE-2021-3129: Unauthenticated RCE in Laravel using Debug Mode")
+                        .setDescription(
+                            "Ignition before 2.5.2, as used in Laravel, allows unauthenticated"
+                                + " remote attackers to execute arbitrary code because of insecure"
+                                + " usage of file_get_contents() and file_put_contents(). This is"
+                                + " exploitable on sites using debug mode with Laravel before"
+                                + " 8.4.3")
+                        .setRecommendation(
+                            "Update Laravel to at least version 8.4.3, and facade/ignition to at"
+                                + " least version 2.5.2.For production systems it is advised to"
+                                + " disable debug mode within the Laravel configuration."))
+                .build());
+  }
+
+  @Test
+  public void detect_notVulnerable_returnsNoVulnerability()
+      throws IOException, InterruptedException {
+    enqueueMockNotVulnerableResponse();
+
+    NetworkService service =
+        NetworkService.newBuilder()
+            .setNetworkEndpoint(
+                forHostnameAndPort(mockWebServer.getHostName(), mockWebServer.getPort()))
+            .setTransportProtocol(TransportProtocol.TCP)
+            .setServiceName("http")
+            .build();
+    TargetInfo targetInfo =
+        TargetInfo.newBuilder()
+            .addNetworkEndpoints(forHostname(mockWebServer.getHostName()))
+            .build();
+
+    DetectionReportList detectionReports = detector.detect(targetInfo, ImmutableList.of(service));
+
+    RecordedRequest detectionReportRequest = mockWebServer.takeRequest();
+    assertThat(detectionReportRequest.getMethod()).isEqualTo("POST");
+    assertThat(detectionReportRequest.getPath()).endsWith("_ignition/execute-solution");
+
+    assertThat(detectionReports.getDetectionReportsList()).isEmpty();
+  }
+
+  private void enqueueMockVulnerableResponse() {
+    MockResponse mockResponse =
+        new MockResponse()
+            .setResponseCode(500)
+            .setBody("file_get_contents(phar://Tsunami_iDontExist");
+    mockWebServer.enqueue(mockResponse);
+  }
+
+  private void enqueueMockNotVulnerableResponse() {
+    MockResponse mockResponse = new MockResponse().setResponseCode(500).setBody("");
+    mockWebServer.enqueue(mockResponse);
+  }
```

Comment:
internal feedback:

consider moving them into the tests, see e.g. https://abseil.io/resources/swe_at_google.2.pdf#page=277